### ISSUE TITLE
Improve MPS qasm method testing and fix failing tests

### ISSUE THREAD

**Issue by chriseclectic** [Qiskit/qiskit-aer#371]
### Informations

- **Qiskit Aer version**: 0.3
- **Python version**: 3.7
- **Operating system**: MacOS

### What is the current behavior?

The MPS simulator should (eventually) be able to support majority of the instructions of the statevector simulator method. Currently tests for the MPS simulator are barebones, so I tried updating it to run the Statevector tests and found many failed or caused segment faults that were not caught by exceptions.

My testing found:

**Tests:**
**Edited: tick for passing**

- [x] QasmCliffordTests
- [x] QasmCliffordTestsWaltzBasis
- [x] QasmCliffordTestsMinimalBasis
- [x] QasmNonCliffordTests
- [x] QasmNonCliffordTestsWaltzBasis
- [x] QasmNonCliffordTestsMinimalBasis
- [x] QasmAlgorithmTests
- [x] QasmAlgorithmTestsWaltzBasis
- [x] QasmAlgorithmTestsMinimalBasis
- [x] QasmUnitaryGateTests
- [x] QasmReadoutNoiseTests
- [x] QasmMeasureTests
- [x] QasmMultiQubitMeasureTests
- [x] QasmResetTests
- [x] QasmPauliNoiseTests
- [x] QasmResetNoiseTests
- [x] QasmConditionalGateTests,
- [x] QasmConditionalUnitaryTests
- [ ] QasmConditionalKrausTests
- [ ] QasmKrausNoiseTests
- [ ] QasmMethodTests
- [ ] QasmInitializeTests

### Steps to reproduce the problem

Run the following (using the `test_qasm_statevector_simulator.py` file as example) commenting out to only execute one set of tests at a time.

```python
class TestQasmMatrixProductStateSimulator(common.QiskitAerTestCase,
                                   QasmMethodTests,
                                   QasmMeasureTests,
                                   QasmMultiQubitMeasureTests,
                                   QasmResetTests,
                                   QasmConditionalGateTests,
                                   QasmConditionalUnitaryTests,
                                   QasmConditionalKrausTests,
                                   QasmCliffordTests,
                                   QasmCliffordTestsWaltzBasis,
                                   QasmCliffordTestsMinimalBasis,
                                   QasmNonCliffordTests,
                                   QasmNonCliffordTestsWaltzBasis,
                                   QasmNonCliffordTestsMinimalBasis,
                                   QasmAlgorithmTests,
                                   QasmAlgorithmTestsWaltzBasis,
                                   QasmAlgorithmTestsMinimalBasis,
                                   QasmUnitaryGateTests,
                                   QasmReadoutNoiseTests,
                                   QasmPauliNoiseTests,
                                   QasmResetNoiseTests,
                                   QasmKrausNoiseTests
                                   ):
    """QasmSimulator matrix product state method tests."""

    BACKEND_OPTS = {
        "seed_simulator": 314159,
        "method": "matrix_product_state"
    }
```

### What is the expected behavior?

* All the tests for supported operations should pass
* Unsupported operations should raise an exception that an instruction is not supported rather than print warnings and return incorrect results.
* Operations should fail gracefully without causing a seg-fault.

### Suggested solutions

Some apparent issues include:

* Measurement giving incorrect results when sampling is disabled (eg see issue #370)
* Conditional instructions giving incorrect result
* Reset instructions causing a seg-fault
* Kraus operations showing as an allowed instruction when they aren't yet implemented.

**Comment by merav-aharoni** [comment]
Following the latest fixes (pending merge of PR #390 and #393), all tests pass but 2:
QasmMethodTests and QasmInitializeTests.

**Comment by chriseclectic** [comment]
@merav-aharoni thats great! The method test isn't too important since we don't select the MPS method automatically, but we should figure out whats going wrong with the initializer.

**Comment by chriseclectic** [comment]
I think we can close this issue since existing features are properly tested now, and can track features that need to be implemented in #391
